src/main/kotlin/no/nav/tiltakspenger/vedtak/rivers/OvergangsstønadMottattRiver.kt

```diff
@@ -0,0 +1,64 @@
+package no.nav.tiltakspenger.vedtak.rivers
+
+import kotlinx.coroutines.runBlocking
+import kotlinx.coroutines.slf4j.MDCContext
+import mu.withLoggingContext
+import no.nav.helse.rapids_rivers.JsonMessage
+import no.nav.helse.rapids_rivers.MessageContext
+import no.nav.helse.rapids_rivers.RapidsConnection
+import no.nav.helse.rapids_rivers.River
+import no.nav.helse.rapids_rivers.asLocalDateTime
+import no.nav.tiltakspenger.vedtak.client.IVedtakClient
+
+internal class OvergangsstønadMottattRiver(
+    private val vedtakClient: IVedtakClient,
+    rapidsConnection: RapidsConnection,
+) : River.PacketListener {
+
+    init {
+        River(rapidsConnection).apply {
+            validate {
+                it.demandAllOrAny("@behov", listOf("overgangsstønad"))
+                it.demandKey("@løsning")
+                it.demandKey("@id")
+                it.demandKey("@behovId")
+                it.requireKey("ident")
+                it.requireKey("journalpostId")
+                it.requireKey("@opprettet")
+                it.interestedIn("@løsning.overgangsstønad")
+            }
+        }.register(this)
+    }
+
+    override fun onPacket(packet: JsonMessage, context: MessageContext) {
+        runCatching {
+            loggBehovVedInngang("overgangsstønad", packet)
+            withLoggingContext(
+                "id" to packet["@id"].asText(),
+                "behovId" to packet["@behovId"].asText(),
+            ) {
+                val ident = packet["ident"].asText()
+                val behovId = packet["@behovId"].asText()
+                val innhentet = packet["@opprettet"].asLocalDateTime()
+                val journalpostId = packet["journalpostId"].asText()
+                val overgangsstønadData =
+                    packet["@løsning.overgangsstønad"].asObject(OvergangsstønadLøsningDTO::class.java)
+
+                runBlocking(MDCContext()) {
+                    vedtakClient.mottaOvergangsstønad(
+                        overgangsstønadDTO = OvergangsstønadDTO(
```

Comment:
Her hopper du bukk over eventuelle feil. Det er jo i og for seg greit, for det er sikkert ikke lagt opp til at faktainnhenteren kan sende noen feil ennå, så sånn sett er det ett fett. Og mtp å sikre bakoverkompatibilitet på rapiden, så er jeg happy camper nå.

Men det kan anmerkes at det øyeblikket man legger til en feil i faktainnhenteren, så vil den bli oversett her - man overfører da en tom liste med perioder, og glemmer feilen. Så kanskje det hadde gjort seg med en sjekk her, så man kunne kastet en exception i de tilfellene?

(I en del andre rivers så sender vi med feilen hele veien til vedtak, og kaster en exception der istedet. Men det trenger du ikke gjøre - det er strengt tatt ett fett hvor man kaster exceptionen. Det er først når man har en plan for å håndtere en feil at det vil være nødvendig å sende den med til vedtak.)

Reply:
Oppdatert koden til å kaste en RuntimeException i de tilfellene feil ikke er null på OvergangsstønadLøsningDTO 👍